kimchi/src/circuits/lookup/lookups.rs

```diff
@@ -390,13 +394,39 @@ impl LookupPattern {
                     })
                     .collect()
             }
+            LookupPattern::Xor => {
+                (0..4)
+                    .map(|i| {
+                        // each row represents for XOR operations
+                        // where l XOR r = o
+                        //
+                        // 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14
+                        // - - - l l l l r r r r  o  o  o  o
```

Comment:
This pattern doesn't look quite right; shouldn't it be
```
// - - - l - - - r - - - o - - -
// - - - - l - - - r - - - o - -
// - - - - - l - - - r - - - o -
// - - - - - - l - - - r - - - o
```

Reply:
Oh! I didn't know how to represent the fact that at each row there was 4 lookups: one for 3 xor 7 = 11, other for 4 xor 8 = 12, other for 5 xor 9 = 13, a final one for 6 xor 10 = 14. But if this is represented as you suggest, then it is exactly the same pattern as ChaCha and then we can remove this one completely 😄 Actually the code in this pattern and that in ChaCha is identical... Yeah, both are doing the same thing!